JS/lib/CRMWebAPI.js

```diff
@@ -1,509 +1,511 @@
-			}
-		}
-		req.onreadystatechange = function () {
-			if (this.readyState == 4 /* complete */ ) {
-				req.onreadystatechange = null;
-				if ((this.status >= 200) && (this.status < 300)) {
-					callback(false, {
-						'response': this.response,
-						'headers': self.parseResponseHeaders(this.getAllResponseHeaders())
-					});
-				} else {
-					callback(true, this)
-				}
-			}
-		};
-		if (['POST', 'PUT', 'PATCH'].indexOf(method) >= 0) {
-			req.send(payload.data);
-		} else {
-			req.send();
-		}
-	};
-	CRMWebAPI.prototype._GetHttpRequestHTTPS = function (config, method, url, payload, callback) {
-		var parsed_url = this.urllib.parse(url);
-		var options = {
-			hostname: parsed_url.hostname,
-			port: 443,
-			path: parsed_url.path,
-			method: method,
-			headers: {
-				"Accept": "application/json",
-				"OData-MaxVersion": "4.0",
-				"OData-Version": "4.0",
-			}
-		}
-		if (['POST', 'PUT', 'PATCH'].indexOf(method) >= 0) {
-			options.headers['Content-Length'] = payload.data.length;
-			options.headers['Content-Type'] = 'application/json';
-		}
-		if (config.callerId) options.headers["MSCRMCallerID"] = config.callerId;
-		if (config.AccessToken != null) options.headers["Authorization"] = "Bearer " + config.AccessToken;
-		if (payload.headers != undefined) {
-			for (var name in payload.headers) {
-				options.headers[name] = payload.headers[name];
-			}
-		}
-		var req = this.https.request(options, function (res) {
-			var body = '';
-			res.setEncoding('utf8');
-			res.on('data', function (chunk) {
-				body += chunk;
-			})
-			res.on('end', function () {
-				if ((res.statusCode >= 200) && (res.statusCode < 300)) {
-					callback(false, {
-						'response': body,
-						'headers': res.headers
-					});
-				} else {
-					callback(true, {
-						'response': body,
-						'headers': res.headers
-					});
-				}
-			});
-		});
-		req.on('error', function (err) {
-			callback(true, err);
-		})
-		if (['POST', 'PUT', 'PATCH'].indexOf(method) >= 0) {
-			req.write(payload.data);
-		}
-		req.end();
-	};
-	CRMWebAPI.prototype._DateReviver = function (key, value) {
-		var a;
-		if (typeof value === 'string') {
-			a = /^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2}(?:\.\d*)?)Z$/.exec(value);
-			if (a) {
-				return new Date(Date.UTC(+a[1], +a[2] - 1, +a[3], +a[4], +a[5], +a[6]));
-			}
-		}
-		return value;
-	};
-	return CRMWebAPI;
-})();
-if (typeof module !== 'undefined' && module.exports) {
-	module.exports = CRMWebAPI;
-	this.CRMWebAPI = CRMWebAPI;
-} else {
-	this.CRMWebAPI = CRMWebAPI;
-}
\ No newline at end of file
+(function (root, factory) {
+    if (typeof define === 'function' && define.amd) {
+        // AMD
+        define(['CRMWebAPI'], factory);
+    } else if (typeof exports === 'object') {
+        // Node, CommonJS-like
+        module.exports = factory(require('https'), require('url'));
+    } else {
+        // Browser globals (root is window)
+        root.returnExports = factory(root.CRMWebAPI);
```

Comment:
This is the one change that may need to be added back.  The UMD has it listed as 
```
root.returnExports = factory(root.CRMWebAPI);
```
but previously it was
```
this.CRMWebAPI = CRMWebAPI;
```

So I'm thinking it should now be
```
root.CRMWebAPI = CRMWebAPI;
// or Both?
root.CRMWebAPI = CRMWebAPI;
root.returnExports= CRMWebAPI;
```